src/main/java/de/tum/in/www1/exerciseapp/domain/MultipleChoiceQuestion.java

```diff
@@ -66,23 +66,30 @@ public ScoringStrategy getScoringStrategy() {
     }
 
     @Override
-    public boolean isAnswerPerfect(SubmittedAnswer submittedAnswer) {
+    public boolean isAnswerCorrect(SubmittedAnswer submittedAnswer) {
         if (submittedAnswer instanceof MultipleChoiceSubmittedAnswer) {
             MultipleChoiceSubmittedAnswer mcAnswer = (MultipleChoiceSubmittedAnswer) submittedAnswer;
+            // iterate through each answer option and compare its correctness with the answer's selection
             for (AnswerOption answerOption : getAnswerOptions()) {
                 boolean isSelected = false;
+                // search for this answer option in the selected answer options
                 for (AnswerOption selectedOption : mcAnswer.getSelectedOptions()) {
                     if (selectedOption.getId().longValue() == answerOption.getId().longValue()) {
+                        // this answer option is selected => we can stop searching
                         isSelected = true;
                         break;
                     }
                 }
```

Comment:
For now it is fine, but I suggest to extract the whole for loop from line 76 to line 82 into its own method isSelected(AnswerOption answerOption) in the class MultipleChoiceSubmittedAnswer. This would make the whole code a lot easier to read and understand.